teps/tools/tep-template.md.template

```diff
@@ -229,6 +229,15 @@ required for these dependencies?
 their own choices?
 -->
 
+### Conformance
+
+<!--
+https://github.com/tektoncd/community/blob/main/design-principles.md#conformance
+
+- Does this proposal require the user to understand how the Tekton API is implemented?
+- Does this proposal introduce additional Kubernetes concepts into the API? If so, is this necessary?
```

Comment:
The [API spec](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md) is pretty sadly out of date; should we include a point about considering what updates may be made to the spec as a result of the TEP?

Reply:
I added a note about this, lmk what you think